### PR TITLE
BSDT bug fix 

### PR DESCRIPTION
**Pull request by bmuniswamy**
When a contact and service field are on the field set and the user is using the Manage Services button on caseman, the service field is now enabled and is a lookup.

# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
~~- [ ] Place holder data is incorporated (Refer to [Case Management Sample Data document](https://quip.com/cbFcAPJF8t0z))~~
~~- [ ] Base axe-core JEST test included for any new LWC (No critical violations)~~
~~- [ ] CRUD/FLS is enforced in Apex~~
~~- [ ] Permission sets are updated to account for CRUD/FLS for new object metadata~~
~~- [ ] All modified classes are unit tested (Apex) at 100% coverage~~
~~- [ ] UX approval or UX not necessary~~
~~- [ ] PR contains draft release notes~~
- [x] Attach work item to the pull request with [Lurch](https://salesforce.quip.com/50ZRA5LEWVzH) `**Lurch:attach W-000000` or `**Lurch:add`
- [ ] All **acceptance criteria** have been met
    - [x] Developer
    - [x] Code Reviewer
    - [ ] QA
- [ ] QE story level testing completed

### CODE REVIEW

**Comment by bmuniswamy** [comment]
**Lurch:attach W-042330

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000007hbzhQAA" target="_blank">W-042330</a>

File: force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js [hunk 1]
```diff
@@ -531,7 +530,7 @@ export default class ServiceDeliveryRow extends LightningElement {
                 }
             });
 
-            if (this.hasContactField && contactId) {
+            if (this.hasContactField && this.hasProgramEngagementField && contactId) {
```
Comment (jjbennett) [inline force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js:533]: @bmuniswamy - do we not need to set selected contact for anything else besides handleGetSserviceEngagements?

Comment (bmuniswamy) [reply force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js:533]: No @jjbennett , I don't think so

**Pull request merged**